core/src/main/kotlin/com/rudderstack/sdk/kotlin/core/internals/queue/EventQueue.kt

```diff
@@ -29,6 +27,7 @@ import java.io.FileNotFoundException
 
 internal const val UPLOAD_SIG = "#!upload"
 private const val BATCH_ENDPOINT = "/v1/batch"
+private val ANONYMOUS_ID_REGEX = """"anonymousId"\s*:\s*"([^"]+)"""".toRegex()
```

Comment:
Add an explanation in the Implementation Detail section regarding this REGEX. Mention what values are allowed and not allowed, providing sample values for both. Also, specify in case of not-allowed values do we fall back to some default value.